src/main/java/kaiyap/KaiYap.java

```diff
@@ -0,0 +1,214 @@
+     *
+     * @param input A string containing the command and index of the task to be deleted.
+     * @throws KaiYapException If the input is missing, invalid, or if the task does not exist.
+     */
+    void deleteTask(String input) throws KaiYapException {
+        if (input.length() <= 7) {
+            throw new MissingInputException("\tSorry, it seems like there is some missing input. Please try again! UwU :3");
+        }
+        int numericIndex = Integer.parseInt(input.substring(7).strip()) - 1;
+        if (numericIndex >= this.taskList.size()) {
+            throw new InvalidInputException("\tSorry, this task does not exist. Please try again! UwU :3");
+        } else {
+            ui.printTaskRemoved(taskList.get(numericIndex), taskList.size() - 1);
+            taskList.remove(numericIndex);
+        }
+    }
+
+    /**
+     * Displays the information about a newly added task.
+     * It prints a message confirming that the task has been added to the list
+     * and shows the total number of tasks currently in the list.
+     *
+     * @param task The task that has been added to the task list.
+     */
+    public void echo(Task task) {
+        System.out.println("\t____________________________________________________________\n" +
+                "\tGot it. I've added this task:\n\t\t" + task.toString() +
+                "\n\tYou now have " + this.taskList.size() + (this.taskList.size() == 1 ? " task" : " tasks") + " in the list.\n" +
+                "\t____________________________________________________________\n");
+    }
+
+    /**
+     * Marks a task as done in the task list based on the user's input.
+     * The input should contain a numeric index that corresponds to the task's position in the list.
+     * If the task is already marked as done, it throws an exception.
+     *
+     * @param index A string containing the command and index of the task to be marked as done.
+     * @throws KaiYapException If the input is missing, invalid, or if the task is already marked as done.
+     */
+    public void markAsDone(String index) throws KaiYapException {
+        if (index.length() <= 5) {
+            throw new MissingInputException("\tSorry, it seems like there is some missing input. Please try again! UwU :3");
+        }
+        int numericIndex = Integer.parseInt(index.substring(5).strip()) - 1;
+        if (numericIndex >= this.taskList.size()) {
+            throw new InvalidInputException("\tSorry, this task does not exist. Please try again! UwU :3");
+        } else if (taskList.get(numericIndex).isTaskDone()) {
+            throw new AlreadyExistsException("\tThis task has already been marked as done. Great job!");
+        } else {
+            taskList.get(numericIndex).setTaskDone(true);
+            System.out.println(
+                    "\t____________________________________________________________\n" +
```

Comment:
Perhaps can move the seperator line into a static variable so no need to copy paste the long line everywhere?

Like 
```java
static String SEPARATOR = ".....\n";
```

Reply:
Great advice, thanks! I've fixed this.